lib/AssemblyUtil/FastaToAssembly.py

```diff
@@ -473,6 +482,18 @@ def _validate_mass_params(self, params):
                 raise ValueError(f"Missing {_ASSEMBLY_NAME} field in {_INPUTS} entry #{i}")
         self._get_int(params.get(_MCL), f"If provided, {_MCL}", minimum=2)
 
+        # validate object_metadata is a mapping if provided
+        obj_metas = [i.get(_OBJ_META) for i in inputs]
+        for i, obj_meta in enumerate(obj_metas, start=1):
+            if obj_meta is not None and type(obj_meta) != dict:
+                raise ValueError(f"{_OBJ_META} must be a mapping if provided for entry #{i}")
+            # validate object_metadata keys and values are strings
+            if obj_meta:
+                if any(not isinstance(v, str) for v in obj_meta.values()):
+                    raise ValueError(f"{_OBJ_META} values must be strings for entry #{i}")
+                if any(not isinstance(k, str) for k in obj_meta):
+                    raise ValueError(f"{_OBJ_META} keys must be strings for entry #{i}")
```

Comment:
This is up to you, but what I do in this situation is put the offending key in the error message to help the user find the problem more quickly.

Reply:
👍

Reply:
The current change is putting the entire object metadata in the error message which could be huge and doesn't really help the user figure out where the error is. I was thinking something like

```python
for key, value in obj_meta.items():
    if not isinstance(key, str):
        raise ValueError(f"{_OBJ_META} key {key} is not a string for entry #{i}")
    if not isinstance(value, str):
        raise ValueError(f"{_OBJ_META} value for key {key} is not a string for entry #{i}")
```

Reply:
👍